api/resolvers/notifications.js

```diff
@@ -106,25 +114,40 @@ export default {
           AND "Item".created_at >= "UserSubscription".created_at
         ${await filterClause(me, models)}
         ORDER BY "sortTime" DESC
-        LIMIT ${LIMIT}+$3)`
+        LIMIT ${LIMIT}+$3`
       )
 
+      // mentions
       if (meFull.noteMentions) {
-        queries.push(
-          `(SELECT "Item".id::TEXT, "Mention".created_at AS "sortTime", NULL as "earnedSats",
+        itemDrivenQueries.push(
+          `SELECT "Item".id::TEXT, "Mention".created_at AS "sortTime", NULL as "earnedSats",
             'Mention' AS type
             FROM "Mention"
             JOIN "Item" ON "Mention"."itemId" = "Item".id
-            LEFT JOIN "Item" p ON "Item"."parentId" = p.id
             WHERE "Mention"."userId" = $1
             AND "Mention".created_at <= $2
             AND "Item"."userId" <> $1
-            AND (p."userId" IS NULL OR p."userId" <> $1)
             ${await filterClause(me, models)}
             ORDER BY "sortTime" DESC
-            LIMIT ${LIMIT}+$3)`
+            LIMIT ${LIMIT}+$3`
         )
       }
+      // Inner union to de-dupe item-driven notifications
+      queries.push(
+        // Only record per item ID
+          `(SELECT DISTINCT ON (id) *
+          FROM (
+            SELECT *
+            FROM (
+              ${itemDrivenQueries.map(q => `(${q})`).join(' UNION ALL ')}
```

Comment:
I removed this `UNION DISTINCT` because the distinct on (id) already covers distinctness